### ISSUE TITLE
No configuration menu elements visible

### ISSUE THREAD

**Issue by cafuneandchill** [cerus/observatorycore-linux#1]
## Description

Nothing is displayed in the Core tab of Obervatory.

### Screenshots

![image](https://user-images.githubusercontent.com/43551330/222356153-2ca64045-b596-4220-bbec-8f2ce2dda430.png)

## Steps to reproduce

1. Open a terminal in the EO root directory.
2. Run `./ObservatoryCore`
3. Open the _Core_ tab.

## System info

```console
$ inxi -b
System:
  Host: echeveria Kernel: 5.15.0-60-lowlatency x86_64 bits: 64
    Desktop: MATE 1.26.0 Distro: Linux Mint 21.1 Vera
Machine:
  Type: Laptop System: Dell product: Dell G15 5510 v: N/A
    serial: <superuser required>
  Mobo: Dell model: 0VKDJ6 v: A00 serial: <superuser required> UEFI: Dell
    v: 1.4.0 date: 06/02/2021
Battery:
  ID-1: BAT0 charge: 84.3 Wh (100.0%) condition: 84.3/84.3 Wh (100.0%)
CPU:
  Info: 8-core Intel Core i7-10870H [MT MCP] speed (MHz): avg: 1009
    min/max: 800/5000
Graphics:
  Device-1: Intel CometLake-H GT2 [UHD Graphics] driver: i915 v: kernel
  Device-2: NVIDIA GA106M [GeForce RTX 3060 Mobile / Max-Q] driver: nvidia
    v: 470.161.03
  Device-3: Microdia Integrated_Webcam_HD type: USB driver: uvcvideo
  Display: x11 server: X.Org v: 1.21.1.3 driver: X:
    loaded: modesetting,nvidia unloaded: fbdev,nouveau,vesa gpu: i915
    resolution: 1920x1080~165Hz
  OpenGL: renderer: NVIDIA GeForce RTX 3060 Laptop GPU/PCIe/SSE2
    v: 4.6.0 NVIDIA 470.161.03
Network:
  Device-1: Intel Comet Lake PCH CNVi WiFi driver: iwlwifi
  Device-2: Realtek driver: r8169
Drives:
  Local Storage: total: 953.88 GiB used: 864.8 GiB (90.7%)
Info:
  Processes: 514 Uptime: 1h 30m Memory: 15.41 GiB used: 6.02 GiB (39.1%)
  Shell: Bash inxi: 3.3.13
```

**Comment by cerus** [comment]
Hey there, are the other two menus displaying anything? What's the output of `dotnet --info`? Where there any messages / errors in the terminal after launching the app?

**Comment by cafuneandchill** [comment]
The other menus just display the normal empty tables. There were no errors in the terminal.

```console
$ dotnet --info
.NET SDK (reflecting any global.json):
 Version:   6.0.113
 Commit:    4a23b50f97

Runtime Environment:
 OS Name:     linuxmint
 OS Version:  21.1
 OS Platform: Linux
 RID:         ubuntu.22.04-x64
 Base Path:   /usr/lib/dotnet/sdk/6.0.113/

global.json file:
  Not found

Host:
  Version:      6.0.13
  Architecture: x64
  Commit:       1af80ba017

.NET SDKs installed:
  6.0.113 [/usr/lib/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.13 [/usr/lib/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.13 [/usr/lib/dotnet/shared/Microsoft.NETCore.App]

Download .NET:
  https://aka.ms/dotnet-download

Learn about .NET Runtimes and SDKs:
  https://aka.ms/dotnet/runtimes-sdk-info
```

**Comment by cerus** [comment]
Thanks for the additional info. I'm currently setting up a Linux Mint virtual machine to reproduce this issue, I'll get back to you shortly

**Comment by cerus** [comment]
Managed to reproduce this issue, working on finding out why it's happening now

**Comment by cerus** [comment]
Update: [Found the issue](https://learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/6.0/system-drawing-common-windows-only), but I'm not yet sure how to fix it. Setting the suggested runtime switch does not seem to have any effect. The app is also working fine on my Arch machine without setting the switch. Could you check if `libgdiplus` is installed? If it isn't please install it and try the program again

**Comment by cafuneandchill** [comment]
```console
$ nala search libgdiplus
libgdiplus 6.0.5-0xamarin1+ubuntu2004b1 [Xamarin-Stable-Focal/focal main]
├── is installed
└── interface library for System.Drawing of Mono
```

**Comment by cerus** [comment]
Alright, I've implemented a workaround for this issue. Please download the latest version of `build.sh` and run `./build.sh cerus linux-fonts-and-cleanup`. If that build works on your machine I'll submit the fix as a pull request to the original repo.

**Comment by cafuneandchill** [comment]
Yep, the Core menu works now. Reads journal logs without issues. Thanks for the help!

One question, though -- where does EO look for plugins on Linux?

**Comment by cerus** [comment]
There's a plugins folder inside of EOs root folder. :)

**Comment by cafuneandchill** [comment]
~~Ah, I see. The Windows version that I run under a Wine prefix looks for plugins in the `AppData/Roaming` folder.~~

Never mind, I'm being dumb.

**Comment by Silarn** [comment]
I would note that I was running into this issue with the latest build on Fedora 28 and installing `libgdiplus` fixed it for me.

**Comment by cerus** [comment]
> I would note that I was running into this issue with the latest build on Fedora 28 and installing `libgdiplus` fixed it for me.

Thanks for your feedback, `libgdiplus` seems to fix this issue on most systems.

**Comment by jimmon89** [comment]
> Alright, I've implemented a workaround for this issue. Please download the latest version of `build.sh` and run `./build.sh cerus linux-fonts-and-cleanup`. If that build works on your machine I'll submit the fix as a pull request to the original repo.

i had the same issue on CachyOS( an arch linux derivative)
installing libgdiplus didn't fix it
but compiling it with the arguments listed here worked

# System Info
```
❯ inxi -b
System:
  Host: cachyos-x8664 Kernel: 6.8.2-2-cachyos-lto arch: x86_64 bits: 64
  Desktop: KDE Plasma v: 6.0.3 Distro: CachyOS
Machine:
  Type: Desktop Mobo: ASUSTeK model: PRIME B450-PLUS v: Rev X.0x
    serial: <superuser required> UEFI: American Megatrends v: 3802
    date: 04/28/2022
CPU:
  Info: quad core AMD Ryzen 5 3400G with Radeon Vega Graphics [MT MCP]
    speed (MHz): avg: 2920 min/max: 1400/3700
Graphics:
  Device-1: NVIDIA GA106 [GeForce RTX 3060] driver: nvidia v: 550.67
  Device-2: Razer USA Gaming Webcam [Kiyo] driver: snd-usb-audio,uvcvideo
    type: USB
  Display: x11 server: X.Org v: 21.1.11 with: Xwayland v: 23.2.4 driver: X:
    loaded: nvidia gpu: nvidia,nvidia-nvswitch resolution: 1: 1920x1080~60Hz
    2: 1920x1080~60Hz 3: 1920x1080~60Hz
  API: OpenGL v: 4.6.0 compat-v: 4.5 vendor: nvidia mesa v: 550.67
    renderer: NVIDIA GeForce RTX 3060/PCIe/SSE2
Network:
  Device-1: Realtek RTL8111/8168/8211/8411 PCI Express Gigabit Ethernet
    driver: r8168
  Device-2: Broadcom BCM4352 802.11ac Dual Band Wireless Network Adapter
    driver: wl
Drives:
  Local Storage: total: 5.69 TiB used: 3.7 TiB (65.0%)
Info:
  Memory: total: 32 GiB available: 31.25 GiB used: 14.54 GiB (46.5%)
  Processes: 438 Uptime: 24m Shell: Zsh inxi: 3.3.33
  ```
  
  # DotNet Info
  ```
  ❯ dotnet --info
.NET SDK:
 Version:           8.0.103
 Commit:            6a90b4b4bc
 Workload version:  8.0.100-manifests.e99a2be4

Runtime Environment:
 OS Name:     cachyos
 OS Version:
 OS Platform: Linux
 RID:         arch-x64
 Base Path:   /usr/share/dotnet/sdk/8.0.103/

.NET workloads installed:
 Workload version: 8.0.100-manifests.e99a2be4
There are no installed workloads to display.

Host:
  Version:      8.0.3
  Architecture: x64
  Commit:       9f4b1f5d66

.NET SDKs installed:
  6.0.128 [/usr/share/dotnet/sdk]
  7.0.117 [/usr/share/dotnet/sdk]
  8.0.103 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.28 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.3 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.28 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.17 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.3 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
  ```
